File: U4/Java/U4P_3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompan.u4p_3;
import java.util.Scanner;
/**
 *
 * @author alfre
 */
public class U4P_3 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese la temperatura en grados Celsius: ");
        String input = scanner.nextLine();

        try {
            int celsius = Integer.parseInt(input);

            int kelvin = celsius + 273;
            int fahrenheit = (celsius * 9 / 5) + 32;

            System.out.println("Temperatura en grados Kelvin: " + kelvin);
            System.out.println("Temperatura en grados Fahrenheit: " + fahrenheit);
        } catch (NumberFormatException e) {
            System.out.println("Por favor, ingrese un valor válido en grados Celsius.");
        } finally {
            // Cerrar el escáner
            scanner.close();
        }
    }
}

File: U4/C#/U4P_3.cs
using System;

namespace MyApp // Note: actual namespace depends on the project name.
{
    internal class U4P_3
    {
        static void Main(string[] args)
        {
            Console.Write("Ingrese la temperatura en grados Celsius: ");
            string input = Console.ReadLine();

            try
            {
                int celsius = int.Parse(input);

                int kelvin = celsius + 273;
                int fahrenheit = (celsius * 9 / 5) + 32;

                Console.WriteLine("Temperatura en grados Kelvin: " + kelvin);
                Console.WriteLine("Temperatura en grados Fahrenheit: " + fahrenheit);
            }
            catch (FormatException)
            {
                Console.WriteLine("Por favor, ingrese un valor válido en grados Celsius.");
            }

        }
    }
}
